doc/source/serve/advanced-guides/multi-app-container.md

```diff
@@ -0,0 +1,175 @@
+(serve-container-runtime-env-guide)=
+# Run Multiple Applications in Different Containers
+
+This section goes over how you can manage dependencies per application in a more isolated manner by running your Serve applications in separate containers with different images.
+
+:::note
+This is an experimental feature and the API is subject to change. If you have additional feature requests or run into issues, please submit them on [Github](https://github.com/ray-project/ray/issues).
+:::
+
+## Install podman
+
+The `container` runtime environment feature uses [Podman](https://podman.io/) to start and run containers. Please follow these steps to install Podman in your environment. For instance, if you are using Kuberay, you need to use an image that has podman installed for the head and worker group specs.
+
+::::{tab-set}
+
+:::{tab-item} Ubuntu 20.10 and newer
+```bash
+sudo apt-get update
+sudo apt-get install podman -y
+```
+:::
+
+:::{tab-item} Ubuntu 20.04 and older
+The podman package is only available in the official repositories for Ubuntu 20.10 and newer. To install podman in Ubuntu 20.04, you need to first add the software repository as a debian source.
+
+```bash
+sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list"
+sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4D64390375060AA4
+sudo apt-get update
+sudo apt-get install podman -y
+```
+:::
+
+:::{tab-item} Other
+See [Podman Installation Instructions](https://podman.io/docs/installation).
+:::
+
+::::
+
+## Run Serve application in a container
+
+In this example, we are going to walk through deploying two applications in separate containers: a whisper model and a Resnet50 image classification model. 
+
+First, let's set up the docker images with the required dependencies. 
+
+:::note
+The Ray version and Python version in the container should match those of the host environment exactly. Note that for Python, the versions must match down to the patch number.
+:::
+
+In `whisper.Dockerfile`:
+* We are running the Whisper model on GPU, so the base image for the Whisper Dockerfile is the latest Ray GPU image, `rayproject/ray:latest-py38-gpu`.
+* We install the package `faster_whisper`, which is a dependency for the whisper model.
+* We then download the source code for the whisper application into `whisper_example.py`.
+Similarly, `resnet.Dockerfile` uses the appropriate base image, installs necessary dependencies, and downloads the resnet model source code.
+Save the following to files named `whisper.Dockerfile` and `resnet.Dockerfile`.
```

Comment:
```suggestion
`whisper.Dockerfile` does the following:
* Uses the latest Ray GPU image, `rayproject/ray:latest-py38-gpu`, so the Whisper model can run on GPUs.
* Installs the package `faster_whisper`, which is a dependency for the whisper model.
* Download the source code for the whisper application into `whisper_example.py`.

Similarly, `resnet.Dockerfile` uses the appropriate base image, installs necessary dependencies, and downloads the resnet model source code.
Save the following to files named `whisper.Dockerfile` and `resnet.Dockerfile`.
```

Reply:
Might be more concise to annotate the dockerfiles with inline comments instead of describing them in prose